### PR TITLE
[PCM-1913] screen recording metadata to use mids in place of track ids

### PR DESCRIPTION
**Pull request by jensengar**

### CODE REVIEW

**Review by djhouseknecht — APPROVED** [review]

File: src/sessions/screen-recording-session-handler.ts [hunk 1]
```diff
@@ -73,6 +73,21 @@ export default class ScreenRecordingSessionHandler extends BaseSessionHandler {
   }
 
   private updateScreenRecordingMetadatas (session: ScreenRecordingMediaSession, metadatas: ScreenRecordingMetadata[]) {
+    metadatas.forEach((meta) => {
+      // adding any here because I don't want to add this to the public interface
+      (meta as any)._trackId = meta.trackId;
```
Comment (davidmorton-gen) [inline src/sessions/screen-recording-session-handler.ts:78]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/MyPureCloud/genesys-cloud-webrtc-sdk/pullRequest?prid=9447336)
```suggestion
      (meta as unknown)._trackId = meta.trackId;
```

File: test/unit/sessions/screen-recording-session-handler.test.ts [hunk 2]
```diff
@@ -153,3 +150,55 @@ describe('updateOutgoingMedia', () => {
...
+        getTransceivers: jest.fn().mockReturnValue(transceivers)
+      }
+    };
+
+    jest.spyOn(utils, 'requestApi').mockResolvedValue(null);
+
+    await handler['updateScreenRecordingMetadatas'](session as any, metas as any);
+
+    expect(metas[0].trackId).toBe('3');
+    expect((metas[0] as any)._trackId).toBe('trackId123');
```
Comment (davidmorton-gen) [inline test/unit/sessions/screen-recording-session-handler.test.ts:199]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/MyPureCloud/genesys-cloud-webrtc-sdk/pullRequest?prid=9447336)
```suggestion
    expect((metas[0] as unknown)._trackId).toBe('trackId123');
```

File: test/unit/sessions/screen-recording-session-handler.test.ts [hunk 3]
```diff
@@ -153,3 +150,55 @@ describe('updateOutgoingMedia', () => {
...
+
+    const session = {
+      pc: {
+        getTransceivers: jest.fn().mockReturnValue(transceivers)
+      }
+    };
+
+    jest.spyOn(utils, 'requestApi').mockResolvedValue(null);
+
+    await handler['updateScreenRecordingMetadatas'](session as any, metas as any);
```
Comment (davidmorton-gen) [inline test/unit/sessions/screen-recording-session-handler.test.ts:196]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/MyPureCloud/genesys-cloud-webrtc-sdk/pullRequest?prid=9447336)
```suggestion
    await handler['updateScreenRecordingMetadatas'](session as unknown, metas as any);
```

File: test/unit/sessions/screen-recording-session-handler.test.ts [hunk 4]
```diff
@@ -153,3 +150,55 @@ describe('updateOutgoingMedia', () => {
...
+
+    expect(metas[0].trackId).toBe('3');
+    expect((metas[0] as any)._trackId).toBe('trackId123');
+
+    expect(metas[1].trackId).toBe('7');
+    expect((metas[1] as any)._trackId).toBe('trackId456');
```
Comment (davidmorton-gen) [inline test/unit/sessions/screen-recording-session-handler.test.ts:202]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/MyPureCloud/genesys-cloud-webrtc-sdk/pullRequest?prid=9447336)
```suggestion
    expect((metas[1] as unknown)._trackId).toBe('trackId456');
```

**Pull request merged**